clippy_lints/src/matches/single_match.rs

```diff
@@ -129,7 +129,7 @@ fn report_single_pattern(cx: &LateContext<'_>, ex: &Expr<'_>, arm: &Arm<'_>, exp
             PatKind::Lit(Expr {
                 kind: ExprKind::Lit(lit),
                 ..
-            }) if lit.node.is_str() => pat_ref_count + 1,
+            }) if lit.node.is_str() | lit.node.is_bytestr() => pat_ref_count + 1,
```

Comment:
Is it intentionally `|` instead of `||` here?

Reply:
sorry for the typo... https://github.com/rust-lang/rust-clippy/pull/13824/commits/7886fee6d9066375ed9eeb7c538381f69c1146e7